llvm/test/lab3/vasilev_ivan/test2.mir

```diff
@@ -0,0 +1,560 @@
+liveins:
+  - { reg: '$xmm0' }
+  - { reg: '$xmm1' }
+  - { reg: '$xmm2' }
+frameInfo:
+  stackSize:       8
+  offsetAdjustment: -8
+  maxAlignment:    16
+  maxCallFrameSize: 0
+fixedStack:
+  - { id: 0, type: spill-slot, offset: -16, size: 8, alignment: 16 }
+stack:
+  - { id: 0, offset: -64, size: 16, alignment: 16 }
+  - { id: 1, offset: -48, size: 16, alignment: 16 }
+  - { id: 2, offset: -32, size: 16, alignment: 16 }
+machineFunctionInfo: {}
+body:             |
+  bb.0 (%ir-block.3):
+    liveins: $xmm0, $xmm1, $xmm2
+  
+    frame-setup PUSH64r killed $rbp, implicit-def $rsp, implicit $rsp
+    CFI_INSTRUCTION def_cfa_offset 16
+    CFI_INSTRUCTION offset $rbp, -16
+    $rbp = frame-setup MOV64rr $rsp
+    CFI_INSTRUCTION def_cfa_register $rbp
+    MOVAPDmr $rbp, 1, $noreg, -48, $noreg, killed renamable $xmm0 :: (store (s128) into %ir.4)
+    MOVAPDmr $rbp, 1, $noreg, -32, $noreg, killed renamable $xmm1 :: (store (s128) into %ir.5)
+    MOVAPDmr $rbp, 1, $noreg, -16, $noreg, killed renamable $xmm2 :: (store (s128) into %ir.6)
+    renamable $xmm0 = MOVAPDrm $rbp, 1, $noreg, -48, $noreg :: (load (s128) from %ir.4)
+    renamable $xmm1 = MOVAPDrm $rbp, 1, $noreg, -32, $noreg :: (load (s128) from %ir.5)
+    renamable $xmm2 = MOVAPDrm $rbp, 1, $noreg, -16, $noreg :: (load (s128) from %ir.6)
+    ; CHECK: renamable $xmm2 = MOVAPDrm $rbp, 1, $noreg, -16, $noreg :: (load (s128) from %ir.6)
+    ; CHECK-NEXT: $xmm0 = VFMADD213PDZ128r $xmm0, $xmm1, $xmm2, implicit $mxcsr
+    ; CHECK-NEXT: $rbp = frame-destroy POP64r implicit-def $rsp, implicit $rsp
+    renamable $xmm0 = nofpexcept MULPDrr killed renamable $xmm0, killed renamable $xmm1, implicit $mxcsr
+    renamable $xmm0 = nofpexcept ADDPDrr killed renamable $xmm0, killed renamable $xmm2, implicit $mxcsr
+    $rbp = frame-destroy POP64r implicit-def $rsp, implicit $rsp
+    CFI_INSTRUCTION def_cfa $rsp, 8
+    RET64 implicit $xmm0
+
+...
+---
+name:            _Z12muladd_test2Dv2_dS_S_
+alignment:       16
+tracksRegLiveness: true
+tracksDebugUserValues: true
+liveins:
+  - { reg: '$xmm0' }
+  - { reg: '$xmm1' }
+  - { reg: '$xmm2' }
+frameInfo:
+  stackSize:       8
+  offsetAdjustment: -8
+  maxAlignment:    16
+  maxCallFrameSize: 0
+fixedStack:
+  - { id: 0, type: spill-slot, offset: -16, size: 8, alignment: 16 }
+stack:
+  - { id: 0, offset: -80, size: 16, alignment: 16 }
+  - { id: 1, offset: -64, size: 16, alignment: 16 }
+  - { id: 2, offset: -48, size: 16, alignment: 16 }
+  - { id: 3, offset: -32, size: 16, alignment: 16 }
+machineFunctionInfo: {}
+body:             |
+  bb.0 (%ir-block.3):
+    liveins: $xmm0, $xmm1, $xmm2
+  
+    frame-setup PUSH64r killed $rbp, implicit-def $rsp, implicit $rsp
+    CFI_INSTRUCTION def_cfa_offset 16
+    CFI_INSTRUCTION offset $rbp, -16
+    $rbp = frame-setup MOV64rr $rsp
+    CFI_INSTRUCTION def_cfa_register $rbp
+    ; CHECK: MOVAPDmr $rbp, 1, $noreg, -64, $noreg, killed renamable $xmm0 :: (store (s128) into %ir.4)
```

Comment:
As I see in description of 3rd test two fmuladd instructions are expected to be made, but I don't see them. It seems that it is due to using MIR generated after regalloc and stack allocation. Could you use MIR generated earlier for simpler input?